### DISCUSSION TITLE
Why I can't disable the Filter > Library function flag for Rust

### DISCUSSION THREAD

**Discussion by Folyd** [compiler-explorer/compiler-explorer#2618] [Q&A]
My link: https://rust.godbolt.org/z/adnK3vTcW
![image](https://user-images.githubusercontent.com/3369694/115998417-b110e600-a619-11eb-9c38-93d5cf5c4aac.png)

**Comment by mattgodbolt** [comment]
It's not supported in this case: we should probably not show it ticked and/or not show it at all. There's no library filtering in rust, currently.

**Reply by Folyd** [reply]
Ok, I see. However, why this link works? https://rust.godbolt.org/z/8Y8Pra

**Reply by mattgodbolt** [reply]
I'm sorry, my Rust knowledge isn't very strong. I don't see how this relates to the library function filter: I see two Rust sources being compared; one using a `match` and one using `if` statements. Can you help me understand what you were expecting, and how this relates to library function filtering? Thanks!

**Reply by Folyd** [reply]
> I see two Rust sources being compared; one using a match and one using if statements.

Exactly. Nothing is special in this case.

Sorry for my misunderstanding about `library function filtering`. Maybe we can give a concise explanation in the UI or somewhere else?
